R/init.R

```diff
@@ -107,11 +105,10 @@ init <- function(data,
                  header = tags$p(),
                  footer = tags$p(),
                  id = character(0)) {
-  logger::log_trace("init initializing teal app with: data ({ class(data)[1] }).")
-  if (is.list(data) && !inherits(data, "teal_data_module")) {
-    checkmate::assert_list(data, names = "named")
-    data <- do.call(teal.data::teal_data, data)
-  }
+  logger::log_trace("init initializing teal app with: data ({ toString(sQuote(class(data))) }).")
```

Comment:
```suggestion
  logger::log_trace("init initializing teal app with { 'class(data)' }.")
```
Class of `data` can now only be `teal_data`, `teal_data_module`, and `list` (#1031), so no need to concatenate complicated classes. Unless we want to allow extensions of `teal_data` but that is a different discussion.

I would also move this until after `data` has passed argument checks. After all, app initiation does not begin until well after that.

Reply:
moved quotes and added brackets

```suggestion
  logger::log_trace("init initializing teal app with: data ('{ class(data) }').")
```